src/activities/ActivityUsageEntity.js

```diff
@@ -418,14 +442,20 @@ export class ActivityUsageEntity extends Entity {
 	}
 
 	async save(activity) {
-		if (typeof activity.dueDate !== 'undefined' &&
-			activity.dueDate !== this.dueDate()) {
-			await this.setDueDate(activity.dueDate);
-		}
-
 		if (typeof activity.isDraft !== 'undefined' &&
 			activity.isDraft !== this.isDraft()) {
 			await this.setDraftStatus(activity.isDraft);
 		}
+
+		const startDateChanged = typeof activity.startDate !== 'undefined';
```

Comment:
Assuming we do decide to only send the changed dates, then I would probably move all this change tracking logic to the `setDates` implementation.

So here just call `this.setDates(activity.startDate, activity.endDate, activity.dueDate)`.

And in `setDates` do something like i did in the assignment implementation.

```
		if (typeof startDate !== 'undefined' && startDate !== this.startDate()) {
			fields.push({ name: 'startDate', value: startDate });
		}

       ...
       if (fields.length > 0) {
       // perform action
       }
```